### ISSUE TITLE
Metro symlink resolver issue with package.json exports

### ISSUE THREAD

**Issue by successx** [microsoft/rnx-kit#1046]
### What happened?

@rnx-kit/metro-resolver-symlinks/lib/resolver cannot resolve module with new 'exports' field in package.json. This also makes metro fail when using with MetroSymlinksResolver. Some npm packages already have this 'exports' field including ramda (v0.28), rambda. 

### Affected Package

@rnx-kit/metro-resolver-symlinks

### Version

0.1.16

### Which platforms are you seeing this issue on?

- [X] Android
- [X] iOS
- [X] macOS
- [ ] Windows

### System Information

```true
error: Package subpath './package.json' is not defined by "exports" in "../node_modules/rambda/package.json"
...
```

### Steps to Reproduce

Use 'resolveModulePath' from '@rnx-kit/metro-resolver-symlinks/lib/resolver' to get rambda, throw above error.
If using 'rambda' within react-native app, run metro and start app also throw error (albeit very cryptic).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tido64** [comment]
Thanks for filing this bug. I am unable to repro this locally:

```
% node
Welcome to Node.js v14.18.2.
Type ".help" for more information.
> const { resolveModulePath } = require("./lib/resolver");
undefined
> resolveModulePath({ originModulePath: "." }, "rambda/package.json")
'../../node_modules/rambda/package.json'
```

Are you sure it is not Metro that is complaining? Prior to 0.67, it will always try to resolve using haste-map first, and only uses this resolver when it fails. Metro itself currently does not support `exports`. There's an open issue here: https://github.com/facebook/metro/issues/670

**Comment by successx** [comment]
Thanks. After digging a bit, looks like this issue is from node itself. require.resolve 'rambda/package.json' throws error on my node (v16.13.1, v14.7.6). Not sure whether Metro or MetroSymlinksResolver throws exception first though. [Special treatment for package.json resolution and exports?](https://github.com/nodejs/node/issues/33460#)

**Comment by afoxman** [comment]
The nodejs issue https://github.com/nodejs/node/issues/33460 has been abandoned, with the last commented saying there isn't likely to be an exception made for package.json when it isn't mentioned in exports.

@tido64 Is there a fix we can/should make? If not, we should close this issue.

**Comment by tido64** [comment]
I think this may have been solved with #1228. Please bump `@rnx-kit/tools-react-native` to [1.2.1](https://github.com/microsoft/rnx-kit/releases/tag/%40rnx-kit%2Ftools-react-native%401.2.1) and try again. If you still have issues, please file a new issues with a repro.
